src/ucp/core/ucp_am.c

```diff
@@ -661,16 +661,15 @@ static ucs_status_t ucp_am_zcopy_multi(uct_pending_req_t *self)
 
     if (req->send.state.dt.offset != 0) {
         ucp_am_fill_middle_header(&mid_hdr, req);
-        return ucp_do_am_zcopy_multi(self, UCP_AM_ID_FIRST, UCP_AM_ID_MIDDLE,
+        return ucp_do_am_zcopy_multi(self, UCP_AM_ID_LAST, UCP_AM_ID_MIDDLE,
                                      NULL, 0ul, &mid_hdr, sizeof(mid_hdr), NULL,
                                      0ul, ucp_am_zcopy_req_complete, 1);
     }
 
     ucp_am_fill_first_header(&first_hdr, req);
 
-    return ucp_do_am_zcopy_multi(self, UCP_AM_ID_FIRST, UCP_AM_ID_MIDDLE,
-                                 &first_hdr, sizeof(first_hdr),
-                                 NULL, sizeof(mid_hdr),
+    return ucp_do_am_zcopy_multi(self, UCP_AM_ID_FIRST, UCP_AM_ID_LAST,
+                                 &first_hdr, sizeof(first_hdr), NULL, 0ul,
```

Comment:
please check calculation of `max_middle` in `ucp_do_am_zcopy_multi` now

Reply:
it is already checked there on `proto_am.inl:406`